### PR TITLE
[test] Fix test java_major_version for JDK 11

### PR DESCRIPTION
**Pull request by mgreau**
We have updated the Docker image to use JDK 11.
This JDK does not contain a `.` in its version

```
docker run -ti --rm --entrypoint=sh docker.elastic.co/elasticsearch/elasticsearch:7.0.0-alpha1-SNAPSHOT

sh-4.2# $JAVA_HOME/bin/java -version
openjdk version "11" 2018-09-25
```
compared with what we had before with the JDK10

```
docker run -ti --rm --entrypoint=sh docker.elastic.co/elasticsearch/elasticsearch:6.4.1
sh-4.2# $JAVA_HOME/bin/java -version
openjdk version "10.0.2" 2018-07-17
```

This commit fixes the test by removing the `.` character.

I have executed `VERSION_QUALIFIER=alpha1 make from-snapshot test-snapshot` and all the tests passed.

### CODE REVIEW

File: tests/test_java.py [hunk 1]
```diff
@@ -3,7 +3,7 @@ def test_java_is_runnable_via_java_home_env_var(host):
 
 
 def test_java_major_version(host):
-    assert 'openjdk version "11.' in host.run('$JAVA_HOME/bin/java -version').stderr
+    assert 'openjdk version "11' in host.run('$JAVA_HOME/bin/java -version').stderr
```
Comment (jasontedor) [inline tests/test_java.py]: This is really brittle. Could we take a different approach please and allow this to match `11((\.0)*\.[1-9][0-9]*)*`? And then we will be good until JDK 12.

Comment (mgreau) [reply tests/test_java.py]: Not sure I follow, I have done some tests and this will work until JDK12 because it tests the first part of the string.

Comment (jasontedor) [reply tests/test_java.py]: I gave a more precise regular expression that will only match JDK builds for JDK 11 that conform to the official version string specification. I don’t like how we had to change this for 10.0.2 to 11 and then got hung up by the fact that the first release in the 11 series does not have a dot after the major version number. With the change I propose, we get a strong assertion, and only have to make a minor update when we go from 11 to 12.

Comment (mgreau) [reply tests/test_java.py]: got it

**Pull request merged**